File: app/src/main/java/moe/notify/animenotifier/AndroidApplication.java
package moe.notify.animenotifier;

import android.app.Application;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.preference.PreferenceManager;

import com.facebook.stetho.Stetho;
import com.freezingwind.animereleasenotifier.BuildConfig;
import com.raizlabs.android.dbflow.config.FlowConfig;
import com.raizlabs.android.dbflow.config.FlowManager;

import moe.notify.animenotifier.helpers.AlarmHelper;
import moe.notify.animenotifier.receiver.AlarmReceiver;
import timber.log.Timber;
import timber.log.Timber.DebugTree;


public class AndroidApplication extends Application {
    // Schedule alarm
    public static void scheduleAlarm(Context context) {
        SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
        int updateInterval = Integer.parseInt(sharedPrefs.getString("updateInterval", "60"));

        AlarmHelper alarmHelper = new AlarmHelper(context, 1000L * 60L * updateInterval, 1000L) {
            @Override
            protected PendingIntent pendingIntent(Context context, int flags) {
                Intent intent = new Intent(context, AlarmReceiver.class);
                return PendingIntent.getBroadcast(context, 0, intent, flags);
            }
        };

        alarmHelper.scheduleUnconditionally();
    }

    @Override
    public void onCreate() {
        super.onCreate();

        // initiate Timber
        if (BuildConfig.DEBUG) {
            Timber.plant(new DebugTree());
        } else {
            // TODO Crashlytics.start(this);
            // TODO Timber.plant(new CrashlyticsTree());
        }

        FlowManager.init(new FlowConfig.Builder(this)
                .openDatabasesOnInit(true).build());

        // enable stetho
        Stetho.initializeWithDefaults(this);
    }


}

File: app/src/main/java/moe/notify/animenotifier/AndroidApplication.kt
package moe.notify.animenotifier

import android.app.Application
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.support.v7.preference.PreferenceManager
import com.facebook.stetho.Stetho
import com.freezingwind.animereleasenotifier.BuildConfig
import com.squareup.leakcanary.LeakCanary
import com.uphyca.stetho_realm.RealmInspectorModulesProvider
import io.realm.Realm
import moe.notify.animenotifier.helpers.AlarmHelper
import moe.notify.animenotifier.receiver.AlarmReceiver
import timber.log.Timber
import timber.log.Timber.DebugTree


class AndroidApplication : Application() {

  override fun onCreate() {
    super.onCreate()
    if (LeakCanary.isInAnalyzerProcess(this)) {
      // This process is dedicated to LeakCanary for heap analysis.
      // You should not init your app in this process.
      return
    }
    LeakCanary.install(this)

    // initiate Timber
    if (BuildConfig.DEBUG) {
      Timber.plant(DebugTree())
    } else {
      // TODO Crashlytics.start(this);
      // TODO Timber.plant(new CrashlyticsTree());
    }

    Realm.init(this)

    // enable stetho
    Stetho.initialize(
        Stetho.newInitializerBuilder(this)
            .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
            .enableWebKitInspector(RealmInspectorModulesProvider.builder(this).build())
            .build())

  }

  companion object {
    // Schedule alarm
    fun scheduleAlarm(context: Context) {
      val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
      val updateInterval = java.lang.Long.parseLong(sharedPrefs.getString("updateInterval", "30"))

      val alarmHelper = object : AlarmHelper(context, Constants.MINUTE * updateInterval,
          Constants.HOUR) {
        override fun pendingIntent(context: Context, flags: Int): PendingIntent {
          val intent = Intent(context, AlarmReceiver::class.java)
          return PendingIntent.getBroadcast(context, 0, intent, flags)
        }
      }

      alarmHelper.scheduleUnconditionally()
    }

    operator fun get(context: Context): AndroidApplication {
      return context.applicationContext as AndroidApplication
    }
  }

}
